### PR TITLE
Remove obsolete field from docker-compose.yml

### PR DESCRIPTION
**Pull request by krysal**
## Description

Every time a `just` command involving docker is run, you get a warning:

CleanShot 2024-04-24 at 18 39 42@2x

According to [documentation](https://docs.docker.com/compose/compose-file/04-version-and-name/), it can be safely removed.

This PR also removes some unused imports from the file defining API URLs.

## Testing Instructions

Compare this branch with `main` looking at the output of the `just api/up`, for example.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
This is related to #4046, and it seems there may be some other work that's necessary in order to close that issue.

**Comment by sarayourfriend** [comment]
@AetherUnbound Can you specify what additional work is needed based on the issue? If this PR works in CI and locally, then I think we're good to go. There was some speculation that the compose file version changes might affect our existing usage, but it seems like they didn't :thinking: we were only using forward compatible features?

**Review by sarayourfriend — APPROVED** [review]
LGTM. CI passing is a good indication we're good to go here. Things work for me locally too, as far as I can tell!

File: api/conf/urls/__init__.py [hunk 1]
```diff
@@ -5,8 +5,6 @@
 https://docs.djangoproject.com/en/4.2/topics/http/urls/
 """
 
-from django.conf import settings
-from django.conf.urls.static import static
```
Comment (dhruvkb) [inline api/conf/urls/__init__.py:9]: Ah yes, unused imports are not caught in `__init__.py`!

**Review by dhruvkb — APPROVED** [review]
Approving because CI passes.

**Comment by AetherUnbound** [comment]
> There was some speculation that the compose file version changes might affect our existing usage, but it seems like they didn't 🤔

This was my only concern, based on the language you used in the other issue.

**Pull request merged**